java/src/com/ibm/streamsx/rest/AbstractStreamingAnalyticsService.java

```diff
@@ -170,10 +170,17 @@ protected void setAuthorization(String authorization) {
         }
     }
     
+    private static String buildStateMetricKey(String state) {
+        return "buildState_" + state + "Time_ms";
+    }
+    
     @Override
     public Result<Job, JsonObject> buildAndSubmitJob(File archive, JsonObject jco,
             String buildName) throws IOException {
-    	
+        
+        JsonObject metrics = new JsonObject();
+        metrics.addProperty("buildArchiveSize", archive.length());
```

Comment:
It might be nice to have an interface with the properties of the metrics object so we can keep track of what gets set where.